### ISSUE TITLE
Max Tweet Length still 140?

### ISSUE THREAD

**Issue by Lz10000** [twitter/twitter-text#231]
Hi, 

I downloaded the latest twitter-text library, 2.0.9 for Java, and I'm calling isValidTweet() from Validator.java and it seems to still check for a max length of 140. I thought the new version would increase the length to 280? 
Thanks,
Liz

**Comment by sudhee** [comment]
If you notice, Validator#isValidTweet is marked deprecated. You should use TwitterTextParser instead:
final TwitterTextParseResults parseResults = TwitterTextParser.parseTweet("tweet text");
parseResults.isValid would return the validity of the tweet.

See https://github.com/twitter/twitter-text/blob/master/java/src/main/java/com/twitter/twittertext/TwitterTextParseResults.java#L10 for more fields on the TwitterTextParseResults object.

**Comment by pawelpredki** [comment]
I have a similar issue. The isValid flag is set to true, the permillage is 500 but the valid range is still 140:

```
public static String getValidTwitterStatusMessage(String userMessage) {
final TwitterTextParseResults result = TwitterTextParser.parseTweet(userMessage);
logger.info(String.format("getValidTwitterStatusMessage2: Permillage <%d> | Valid <%d> | Weighted length <%d> | Range <%d-%d>", result.permillage, result.isValid ? 1 : 0, result.weightedLength, result.validTextRange.start, result.validTextRange.end));

if (result.permillage > 1000) {
	logger.warning("getValidTwitterStatusMessage2: message too long <"
		+ result.permillage + ">");
}
return userMessage.substring(result.validTextRange.start,
		result.validTextRange.end + 1);
}
```

```
getValidTwitterStatusMessage2: Permillage <500> | Valid <1> | Weighted length <140> | Range <0-139>
```

I'm doing a permillage check and it seems not to go beyond 1000.

I'm using this as a test input:

```
1234567890abcdefghij1234567890abcdefghij1234567890abcdefghij1234567890abcdefghij1234567890abcdefghij1234567890abcdefghij1234567890abcdefghij1234567890abcdefghij
```

**Comment by denisk20** [comment]
> If you notice, Validator#isValidTweet is marked deprecated. You should use TwitterTextParser instead

Deprecation doesn't mean that the component should be broken. It just indicates that users are urged to switch to a newer component and that the old one will be removed soon. `Validator` is clearly broken at the moment.

**Comment by sudhee** [comment]
This is fixed in 2.0.10 & #243
